tests/Resources/ti.blob.test.js

```diff
@@ -0,0 +1,337 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions'),
+	utilities = require('./utilities/utilities');
+
+describe.only('Titanium.Blob', function () {
+	var win;
+
+	afterEach(function () {
+		if (win) {
+			win.close();
+		}
+		win = null;
+	});
+
+	it('.apiName', function () {
+		// FIXME Should be able to do Ti.Blob.apiName
+		var blob = Ti.Filesystem.getFile('app.js').read();
+		should(blob).have.a.readOnlyProperty('apiName').which.is.a.String;
+		should(blob.apiName).be.eql('Ti.Blob');
+	});
+
+	it('constructed from File.read()', function () {
+		var blob = Ti.Filesystem.getFile('app.js').read();
+		should(blob).be.an.Object;
+		// should(blob).be.an.instanceof(Ti.Blob); // Crashes Windows, throws uncaught error on iOS & Android
+	});
+
+	// Windows crashes on instanceof check TIMOB-25012
+	// Windows also crashes if we uncomment this now, I think closing the window (or failing the test) in the blob callback is causing Desktop crash
+	// Android is sometimes timing out... Trying an open event now...
+	// TODO: Test is tempermental, skipping for now...
```

Comment:
Can be removed as it seems like it's not being skipped (anymore).

Reply:
I uncommented/unskipped all tests to see if I could fix them all. This one still needs to be skipped as Android times out.

Reply:
@sgtcoolguy, I suspect the Android timeout is caused by opening a window while a previous windows is closing. Not sure why yet (haven't had time to investigate) but doing the following in the `aftereach()` call resolves it for me. Trouble is you would have to implement this in all unit test files.

```javascript
afterEach(function (finish) {
	if (win) {
		win.addEventListener("close", function () {
			finish();
		});
		win.close();
		win = null;
	} else {
		finish();
	}
});
```